### PR TITLE
Add missing test for credential provider.

### PR DESCRIPTION
**Pull request by dlaidlaw**
The credential provider for AWS CodeCommit cannot be executed without the change in the setCredentialsProvider method of JGitEnvironmentRepository.java. This test got dropped in the original pull request. The original pull request was #548

### CODE REVIEW

File: spring-cloud-config-client/.gitignore [hunk 1]
```diff
@@ -0,0 +1 @@
+/.apt_generated/
```
Comment (ryanjbaxter) [inline spring-cloud-config-client/.gitignore]: Is this needed?

Comment (dlaidlaw) [inline spring-cloud-config-client/.gitignore]: It was generated somewhere. I did not add it. I googled and it says that that is where code generated by Annotations is placed. And I do believe that there is code generated by Lombok in there. So it is probably a good idea, but I can remove it if that is preferable. There are a couple of .apt_generated folders generated by the build on my machine.

Comment (ryanjbaxter) [inline spring-cloud-config-client/.gitignore]: Hum, it seems odd because nothing has changed.  I think it should be removed.

Comment (dlaidlaw) [inline spring-cloud-config-client/.gitignore]: I removed it, and added some tests to improve code coverage, and test that the change actually performed as expected.

**Pull request merged**